main/src/ca/uwaterloo/flix/language/phase/Instances.scala

```diff
@@ -169,4 +190,14 @@ object Instances extends Phase[TypedAst.Root, TypedAst.Root] {
       case ((), x) => f(x)
     }
   }
+
+  /**
+    * Returns true iff `tpe1` unifies with `tpe2`.
+    */
+  private def unifiesWith(tpe1: Type, tpe2: Type)(implicit flix: Flix): Boolean = {
```

Comment:
I would this into the top of `Unification.scala`. Easier to maintain and discovered if needed elsewhere.